pkg/backend/storage/fs/store.go

```diff
@@ -54,7 +55,25 @@ func (s *Store) Set(ctx context.Context, name string, value []byte) error {
 func (s *Store) Delete(ctx context.Context, name string) error {
 	path := filepath.Join(s.path, filepath.Clean(name))
 	out.Debug(ctx, "fs.Delete(%s) - %s", name, path)
-	return os.Remove(path)
+
+	err := os.Remove(path)
+	if err != nil {
+		return err
+	}
+
+	// remove all empty parent directories
+	for dir := filepath.Dir(path); ; dir = filepath.Dir(dir) {
```

Comment:
The three form for is uncommon in Go. Please add a comment describing what it does (e.g. "start with parent, then work upwards, ...").

Reply:
:+1:

Reply:
With the other requested changes it makes much more sense to turn this loop into a recursive function. It should be much more readable now